Define ET_HAS_EXCEPTIONS macro

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #11604
* __->__ #11603

To support passing ET_USE_PYTORCH_HEADERS only when exceptions are enabled.

Differential Revision: [D76470039](https://our.internmc.facebook.com/intern/diff/D76470039/)

swolchok committed:
Define ET_HAS_EXCEPTIONS macro

To support passing ET_USE_PYTORCH_HEADERS only when exceptions are enabled.

Differential Revision: [D76470039](https://our.internmc.facebook.com/intern/diff/D76470039/)

[ghstack-poisoned]

JacobSzwejbka commented on runtime/platform/compiler.h:
```diff
@@ -171,6 +171,14 @@
 
+#ifdef __EXCEPTIONS
+#define ET_HAS_EXCEPTIONS 1
+#elif defined(_HAS_EXCEPTIONS) && _HAS_EXCEPTIONS
```
Oh its in the test plan. From the compilers.

swolchok committed:
Update on "Define ET_HAS_EXCEPTIONS macro"


To support passing ET_USE_PYTORCH_HEADERS only when exceptions are enabled.

Differential Revision: [D76470039](https://our.internmc.facebook.com/intern/diff/D76470039/)

[ghstack-poisoned]
